packages/@headlessui-react/src/components/combobox/combobox.tsx

```diff
@@ -821,12 +821,19 @@ function InputFn<
   )
 
   let isComposing = useRef(false)
+  let composedChangeEvent = useRef<React.ChangeEvent<HTMLInputElement> | null>(null)
   let handleCompositionStart = useEvent(() => {
     isComposing.current = true
   })
   let handleCompositionEnd = useEvent(() => {
     setTimeout(() => {
```

Comment:
I think that changing this to `d.nextFrame` solves the odd issue you are seeing when pressing <kbd>enter</kbd>

Reply:
hmm I'll give that a try